src/core/comment.ml

```diff
@@ -0,0 +1,153 @@
+(*
+ * Module to manage commentaries
+ * DURAND - MARAIS © 2019
+ *)
+
+
+open Monad
+open Ezjsonm
+
+
+
+(* Module to manage string map *)
+module Map_Str = Map.Make(String)
+
+(* Type commentary *)
+type comment = 
+  | Block of string*string*string
+  | Inline of string
+
+(* Auxiliary constructors *)
+let inline s = Inline s
+let block (s1,s2,s3) = Block (s1,s2,s3)
+
+
+
+(* default commentary list *)
+let default = [
+  ["c" ; "java" ; "go"], Block ("/*", " * ", " */")  ;
+  ["py" ; "sh"], Inline "#" ;
+  ["ml" ; "mli"], Block ("(*", " * ", " *)")  ;
+  [".vim"], Inline "\"";
+]
+
+(* Return an inline value *)
+let get_inline v : comment choice =
+  match v with
+  | `String s -> Ok (inline s)
+  | _ -> Error "[Error] Inline needs a String value"
+
+(* Return a block value *)
+let get_block v =
+  (
+    try Ok (get_list get_string v)
+    with _ -> Error "[Error] Block needs an array of strings"
+  ) >>=
+  (
+    fun l ->
+      try Ok (List.nth l 0 , List.nth l 1, List.nth l 2)
+      with _ -> Error "[Error] Block needs exactly three strings"
+  ) >>= (fun t ->  Ok (block t))
+
+(* Get ext value *)
+let get_ext v =
+  let rec aux l =
+    match l with
+    | [] -> Error "[Error] Can't get field ext"
+    | ("ext", value)::_ -> (
+        try Ok (get_list get_string value)
+        with _ -> Error "[Error] Ext needs a list of strings"
+      )
+    | _::q -> aux q
+  in aux v
+
+(* Build get extension content *)
+let get_ext_type v =
+  let rec aux l =
+    match l with
+    | [] -> Error "[Error] Can't find inline or block value"
+    | ("inline", value)::_ -> get_inline value
+    | ("block", value)::_ -> get_block value
+    | _::q -> aux q 
+  in aux v 
+
+(* Create a comment *)
+let create_com v =
+  let ext = get_ext v in
+  let ext_type = get_ext_type v in
+  match ext, ext_type with
+  | Ok e, Ok t -> Ok (e, t)
+  | Ok _, Error a -> Error a
+  | Error a, _ -> Error a
+
+
+
+(* Build a default list *)
+let build_map lst =
+  let rec from_list lst v m =
```

Comment:
You could use List.fold_left.

Reply:
Replace but let it in a function to be more readable :)